test/components/linkerd/linkerd_test.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build aws aks aws_edge packet
+// +build e2e
+
+package linkerd_test
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/yaml"
+
+	testutil "github.com/kinvolk/lokomotive/test/components/util"
+)
+
+const (
+	namespace = "linkerd"
+)
+
+func TestLinkerdDeployment(t *testing.T) {
+	deployments := []string{
+		"linkerd-controller",
+		"linkerd-destination",
+		"linkerd-grafana",
+		"linkerd-identity",
+		"linkerd-prometheus",
+		"linkerd-proxy-injector",
+		"linkerd-sp-validator",
+		"linkerd-tap",
+		"linkerd-web",
+	}
+
+	client := testutil.CreateKubeClient(t)
+
+	for _, d := range deployments {
+		d := d
+		t.Run(d, func(t *testing.T) {
+			t.Parallel()
+
+			testutil.WaitForDeployment(t, client, namespace, d, retryInterval, timeout)
+		})
+	}
+}
+
+const (
+	// nolint: lll
+	podConfig = `
+apiVersion: v1
+kind: Pod
+metadata:
+  generateName: linkerd-check-test-
+spec:
+  restartPolicy: Never
+  containers:
+  - image: fedora
+    name: linkerd-cli
+    env:
+    - name: KUBECONFIG
+      value: /root/.kube/config
+    command:
+    - bash
+    args:
+    - -c
+    - 'curl -sLO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl && chmod +x ./kubectl &&  mv ./kubectl /usr/local/bin/ && kubectl version -o yaml && curl -sL https://run.linkerd.io/install | sh && export PATH=$PATH:$HOME/.linkerd2/bin && linkerd version && linkerd check'
```

Comment:
Maybe we can use init containers to download things to empty volume and then have only last commands here? That would make it more readable.